part-2.py

```diff
@@ -4,13 +4,27 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if not array:
+        return False
+    else:
+        return array[0] == query or search(array[1:], query)
```

Comment:
This is a valid recursive solution, but it's going to do a lot of extra work.  One base case is that we can't search anymore because there's nothing left in the array, but the other base case is we found the query.